flaky test duration tiflash test

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2022/12/14 23:22:34.788 +08:00] [ERROR] [ddl_tiflash_api.go:396] ["get tiflash sync progress failed"] [error="Get \"http://127.0.0.1:40341/tiflash/sync-status/81\": dial tcp 127.0.0.1:40341: connect: connection refused"] [errorVerbose="Get \"http://127.0.0.1:40341/tiflash/sync-status/81\": dial tcp 127.0.0.1:40341: connect: connection refused\ngithub.com/pingcap/errors.AddStack\n\texternal/com_github_pingcap_errors/errors.go:174\ngithub.com/pingcap/errors.Trace\n\texternal/com_github_pingcap_errors/juju_adaptor.go:15\ngithub.com/pingcap/tidb/store/helper.CollectTiFlashStatus\n\tstore/helper/helper.go:1222\ngithub.com/pingcap/tidb/domain/infosync.getTiFlashPeerWithoutLagCount\n\tdomain/infosync/tiflash_manager.go:91\ngithub.com/pingcap/tidb/domain/infosync.calculateTiFlashProgress\n\tdomain/infosync/tiflash_manager.go:117\ngithub.com/pingcap/tidb/domain/infosync.(*mockTiFlashReplicaManagerCtx).CalculateTiFlashProgress\n\tdomain/infosync/tiflash_manager.go:752\ngithub.com/pingcap/tidb/domain/infosync.CalculateTiFlashProgress\n\tdomain/infosync/info.go:1042\ngithub.com/pingcap/tidb/ddl.pollAvailableTableProgress\n\tddl/ddl_tiflash_api.go:394\ngithub.com/pingcap/tidb/ddl.(*ddl).refreshTiFlashTicker\n\tddl/ddl_tiflash_api.go:435\ngithub.com/pingcap/tidb/ddl.(*ddl).PollTiFlashRoutine\n\tddl/ddl_tiflash_api.go:574\ngithub.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1\n\tutil/wait_group_wrapper.go:33\nruntime.goexit\n\tGOROOT/src/runtime/asm_amd64.s:1594"] [tableID=81] [IsPartition=true] [stack="github.com/pingcap/tidb/ddl.pollAvailableTableProgress\n\tddl/ddl_tiflash_api.go:396\ngithub.com/pingcap/tidb/ddl.(*ddl).refreshTiFlashTicker\n\tddl/ddl_tiflash_api.go:435\ngithub.com/pingcap/tidb/ddl.(*ddl).PollTiFlashRoutine\n\tddl/ddl_tiflash_api.go:574\ngithub.com/pingcap/tidb/util.(*WaitGroupWrapper).Run.func1\n\tutil/wait_group_wrapper.go:33"]
2022/12/14 23:22:37 httptest.Server blocked in Close after 5 seconds, waiting for connections:
  *net.TCPConn 0xc006a260f8 127.0.0.1:33024 in state active
[2022/12/14 23:22:44.093 +08:00] [INFO] [job_manager.go:257] ["scale ttl worker"] [ttl-worker=manager] [originalCount=0] [newCount=4]
[2022/12/14 23:22:44.093 +08:00] [INFO] [job_manager.go:257] ["scale ttl worker"] [ttl-worker=manager] [originalCount=0] [newCount=4]
-- Test timed out at 2022-12-14 23:30:53 CST --
================================================================================
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/ddl_tiflash_api.go
+++ b/ddl/ddl_tiflash_api.go
@@ -37,6 +37,7 @@ import (
 	"github.com/pingcap/tidb/store/helper"
 	"github.com/pingcap/tidb/table"
 	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/intest"
 	"github.com/pingcap/tidb/util/logutil"
 	atomicutil "go.uber.org/atomic"
 	"go.uber.org/zap"
@@ -398,6 +399,10 @@ func pollAvailableTableProgress(schemas infoschema.InfoSchema, ctx sessionctx.Co
 				zap.Int64("tableID", availableTableID.ID),
 				zap.Bool("IsPartition", availableTableID.IsPartition),
 			)
+			if intest.InTest {
+				// https://github.com/pingcap/tidb/issues/39949
+				panic(err)
+			}
 			continue
 		}
 		err = infosync.UpdateTiFlashProgressCache(availableTableID.ID, progress)

--- a/ddl/ddl_tiflash_api.go
+++ b/ddl/ddl_tiflash_api.go
@@ -400,7 +400,10 @@ func pollAvailableTableProgress(schemas infoschema.InfoSchema, ctx sessionctx.Co
 				zap.Bool("IsPartition", availableTableID.IsPartition),
 			)
 			if intest.InTest {
-				// https://github.com/pingcap/tidb/issues/39949
+				// In the test, the server cannot start up because the port is occupied.
+				// Although the port is random. so we need to quickly return when to
+				// fail to get tiflash sync.
+				//https://github.com/pingcap/tidb/issues/39949
 				panic(err)
 			}
 			continue

--- a/ddl/ddl_tiflash_api.go
+++ b/ddl/ddl_tiflash_api.go
@@ -403,7 +403,7 @@ func pollAvailableTableProgress(schemas infoschema.InfoSchema, ctx sessionctx.Co
 				// In the test, the server cannot start up because the port is occupied.
 				// Although the port is random. so we need to quickly return when to
 				// fail to get tiflash sync.
-				//https://github.com/pingcap/tidb/issues/39949
+				// https://github.com/pingcap/tidb/issues/39949
 				panic(err)
 			}
 			continue

--- a/ddl/tiflashtest/BUILD.bazel
+++ b/ddl/tiflashtest/BUILD.bazel
@@ -7,6 +7,7 @@ go_test(
         "main_test.go",
     ],
     flaky = True,
+    shard_count = 30,
     deps = [
         "//config",
         "//ddl",

--- a/ddl/tiflashtest/BUILD.bazel
+++ b/ddl/tiflashtest/BUILD.bazel
@@ -2,6 +2,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
 go_test(
     name = "tiflashtest_test",
+    timeout = "short",
     srcs = [
         "ddl_tiflash_test.go",
         "main_test.go",

